builtin/providers/azure/provider.go

```diff
@@ -0,0 +1,83 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a teraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"publish_settings_file": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: parameterDescriptions["publish_settings_file"],
+			},
+			"subscription_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["subscription_id"],
+			},
+			"management_certificate": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["management_certificate"],
+			},
+			"management_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["management_url"],
+			},
+		},
+
+		ResourcesMap: map[string]*schema.Resource{
+			"azure_instance":                 resourceAzureInstance(),
+			"azure_hosted_service":           resourceAzureHostedService(),
```

Comment:
I thought about this again, but did you actually test with your current approach? Because I don't think this will work as you might expect. So how would you share a hosted service between multiple instances?